source/server/configuration_impl.h

```diff
@@ -56,6 +56,40 @@ class HttpTracerFactory {
 };
 
 /**
+ * Implemented for each Stats::Sink and registered via Registry::registerFactory() or
+ * the convenience class RegisterFactory.
+ */
+class StatsSinkFactory {
+public:
+  virtual ~StatsSinkFactory() {}
+
+  /**
+   * Create a particular Stats::Sink implementation. If the implementation is unable to produce a
+   * Stats::Sink with the provided parameters, it should throw an EnvoyException in the case of
+   * general error or a Json::Exception if the json configuration is erroneous. The returned
```

Comment:
Why `Json::Exception`? This is a pure v2 feature, we shouldn't have RapidJSON involved as we're working in proto land here. The only JSON conversion is performed by the protobuf library that won't create these types of exceptions.

Reply:
Oh, I see, this is copy+paste from https://github.com/lyft/envoy/blob/master/include/envoy/server/filter_config.h. In `filter_config.h`, the JSON stuff doesn't apply to the `...FromProto()` variants.

Reply:
Right, sorry, that was just a typo.  Thought I removed it.